param/parameterized.py

```diff
@@ -1688,6 +1692,27 @@ def __exit__(self, exc_type, exc_value, exc_tb):
             self._restore = {}
 
 
+def _instantiate_param_obj(paramobj, owner=None):
+    """Return a Parameter object suitable for instantiation given the class's Parameter object"""
+
+    # Shallow-copy Parameter object, with special handling for watchers
+    # (from try/except/finally in Parameters.__getitem__ in https://github.com/holoviz/param/pull/306)
+    p = paramobj
+    watchers = p.watchers
+    p.watchers = {}
+    p = copy.copy(p)
+
+    p.watchers = copy.copy(watchers)
```

Comment:
For context here is the [PR](https://github.com/holoviz/param/pull/419) that changed `p.watchers = watchers` to `p.watchers = {k: list(v) for k, v in watchers.items()}`. @philippjfr if by any chance you remember how to reproduce the error mentioned in the PR we could write a test for it.

@jbednar I'm also unsure why a dict comprehension was used instead of simpler approaches like `watchers.copy()`.

Reply:
The problem is that both the dictionary itself and the `list` values need to be copied but we do not want to copy the `watcher` objects that are already there.

Reply:
What was wrong with the dictionary comprehension?

Reply:
copy.copy is shallow, and therefore doesn't copy the underlying objects, right?

Reply:
I can try, watcher's were originally Chris' thing so I frequently have re-familiarize myself as well.